### PR TITLE
Additional filter operations: isnull and isnotnull

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
This superscedes: https://github.com/json-api-dotnet/JsonApiDotNetCore/pull/386/files
Closes #{ISSUE_NUMBER}

#### FEATURE
- [x] write tests that address the requirements outlined in the issue
- [x] fulfill the feature requirements
- [ ] bump package version

/cc @milosloub

### CODE REVIEW

File: test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs [hunk 1]
```diff
@@ -90,6 +91,52 @@ public async Task Can_Filter_TodoItems()
...
+            var route = $"/api/v1/todo-items?filter[updated-date]=isnot:null";
+            var request = new HttpRequestMessage(httpMethod, route);
+
+            // Act
+            var response = await _fixture.Client.SendAsync(request);
+            var body = await response.Content.ReadAsStringAsync();
+            var deserializedBody = _fixture.GetService<IJsonApiDeSerializer>().DeserializeList<TodoItem>(body);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs]: I getchya. I was just copying from other tests ;)

File: src/JsonApiDotNetCore/Extensions/IQueryableExtensions.cs [hunk 2]
```diff
@@ -113,31 +113,56 @@ public static IQueryable<TSource> Filter<TSource>(this IQueryable<TSource> sourc
...
 
                     return source.Where(lambdaIn);
                 }
+                else if (op == FilterOperations.@is || op == FilterOperations.isnot) {
+                    var parameter = Expression.Parameter(concreteType, "model");
+                    // {model.Id}
+                    var left = Expression.PropertyOrField(parameter, property.Name);
+                    var right = Expression.Constant(filterQuery.PropertyValue, typeof(string));
+
+                    var body = GetFilterExpressionLambda(left, right, op);
```
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply src/JsonApiDotNetCore/Extensions/IQueryableExtensions.cs]: > Maybe something like this. And any value after the operator is ignored.

This is actually what I started with, should be easy to go back. :)

> [Attr(unsetFilterValueBehavior: UnsetFilterValueBehavior.AsNull)]

I guess the case I keep thinking of is what if someone has a nullable string field?


In previously projects (rails), I used https://github.com/activerecord-hackery/ransack which has a bunch of matchers: https://github.com/activerecord-hackery/ransack#search-matchers including null and not_null

File: test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (NullVoxPopuli) [reply test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs]: @jaredcnance how much refactoring would you want on acceptance tests? I typically extract all the repetitive stuff like request building and such into helper methods (I could do this in a subsequent PR)

File: test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs [hunk 3]
```diff
@@ -90,6 +91,56 @@ public async Task Can_Filter_TodoItems()
...
+            // Act
+            var response = await _fixture.Client.SendAsync(request);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var body = await response.Content.ReadAsStringAsync();
+            var deserializedBody = _fixture.GetService<IJsonApiDeSerializer>().DeserializeList<TodoItem>(body);
+
+            // Assert
+            Assert.NotEmpty(deserializedBody);
```
Comment (jaredcnance) [inline test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs]: Let's verify the returned `TodoItem`s have actually been filtered (e.g. all results have `UpdatedDate = null`. 
Maybe even consider explicitly adding a `TodoItem` to the db with `UpdatedDate = notNullDateValue;`
It shouldn't be an issue after #294, but for now I think we should assume we're not working with an empty db.

File: test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs [hunk 4]
```diff
@@ -90,6 +91,56 @@ public async Task Can_Filter_TodoItems()
...
```
Comment (jaredcnance) [inline test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs]: See comment below.

Comment (NullVoxPopuli) [reply test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs]: added assertion

File: test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs (hunk 3 above)
Comment (NullVoxPopuli) [reply test/JsonApiDotNetCoreExampleTests/Acceptance/TodoItemsControllerTests.cs]: added assertion

**Pull request merged**